### PR TITLE
Bump to .NET 6.0.100-preview.3.21161.23

### PR DESCRIPTION
**Pull request by jonathanpeppers**
At first, Android "Hello World" fails with:

    > dotnet build -c Release
    ILLink : error IL1009: Assembly 'System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' reference 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' could not be resolved

Dropping the `--skip-unresolved false` switch passed to ILLink solves
this issue. This was a recommendation from Sven Boemer on the .NET team.

A nice, side-effect, is we no longer need workarounds like this any
longer:

    <ItemGroup Condition=" '$(Configuration)' == 'Release' ">
      <PackageReference Include="System.CodeDom" Version="6.0.0-*" />
      <PackageReference Include="System.Configuration.ConfigurationManager" Version="6.0.0-*" />
      <PackageReference Include="System.Diagnostics.PerformanceCounter" Version="6.0.0-*" />
      <PackageReference Include="System.Diagnostics.EventLog" Version="6.0.0-*" />
      <PackageReference Include="System.Drawing.Common" Version="6.0.0-*" />
      <PackageReference Include="System.IO.Ports" Version="6.0.0-*" />
      <PackageReference Include="System.Threading.AccessControl" Version="6.0.0-*" />
    </ItemGroup>

I removed these from several tests.

`BuildReleaseArm64` test, net6 apk size difference before/after

Simple XA:

```diff
-"PackageSize": 2885411
+"PackageSize": 2889606
```

XF/XA

```diff
-"PackageSize": 9886918
+"PackageSize": 8746124
```

The Xamarin.Forms app saw quite a bit of improvement, since we don't
have to bring in these `@(PackageReference)` anymore.

## Other changes ##

I updated `PerformanceTest` to ignore `NotSupportedException` due to:

    System.NotSupportedException : Unsupported log file format. Latest supported version is 10, the log file has version 11.

We need to use a new version of this package when it is available:

https://www.nuget.org/packages/MSBuild.StructuredLogger/2.1.303

### CODE REVIEW

File: src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml [hunk 1]
```diff
@@ -3,5 +3,6 @@
 <linker>
 	<assembly fullname="System.Private.CoreLib">
 		<type fullname="System.Globalization.GlobalizationMode" />
+		<type fullname="System.Runtime.CompilerServices.InternalsVisibleToAttribute" />
```
Comment (jonathanpeppers) [inline src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml:6]: This fixed a crash in Mono.Android-Tests:

    Can't find custom attr constructor image: Mono.Android.dll mtoken: 0x0a000464 due to:
    Could not resolve type with token 010000fb from typeref (expected class 'System.Runtime.CompilerServices.InternalsVisibleToAttribute' in assembly 'System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e') assembly:System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e type:System.Runtime.CompilerServices.InternalsVisibleToAttribute member:(null)

File: Configuration.props [hunk 2]
```diff
@@ -77,9 +77,9 @@
     <!-- Version number from: https://github.com/dotnet/installer#installers-and-binaries -->
     <!-- Please update DotNetRuntimePacksVersion below accordingly -->
     <DotNetPreviewVersionBand Condition=" '$(DotNetPreviewVersionBand)' == '' ">6.0.100</DotNetPreviewVersionBand>
-    <DotNetPreviewVersionFull Condition=" '$(DotNetPreviewVersionFull)' == '' ">$(DotNetPreviewVersionBand)-preview.2.21155.3</DotNetPreviewVersionFull>
+    <DotNetPreviewVersionFull Condition=" '$(DotNetPreviewVersionFull)' == '' ">$(DotNetPreviewVersionBand)-preview.3.21161.23</DotNetPreviewVersionFull>
```
Comment (jonathanpeppers) [inline Configuration.props:80]: FYI @rolfbjarne / @spouliot looks Android was able to bump to this version with just some minor changes.

Comment (rolfbjarne) [reply Configuration.props:80]: Yeah, we have a PR going here: https://github.com/xamarin/xamarin-macios/pull/10772, I was just waiting for P2 to go out first.

**Review by dellis1972 — APPROVED** [review]

**Pull request merged**